### PR TITLE
KIALI-2893 Show the traces tab if jaeger is configured but applicatio…

### PR DESCRIPTION
**Pull request by aljesusg**
JIRA: [KIALI-2893](https://issues.jboss.org/browse/KIALI-2893)

Need backend: https://github.com/kiali/kiali/pull/1089

### CODE REVIEW

File: src/pages/ServiceDetails/ServiceDetailsPage.tsx [hunk 1]
```diff
@@ -272,7 +282,6 @@ class ServiceDetails extends React.Component<ServiceDetailsProps, ServiceDetails
                 <div>Inbound Metrics</div>
               </NavItem>
               {errorTraces !== undefined &&
-                errorTraces > -1 &&
```
Comment (hhovsepy) [inline src/pages/ServiceDetails/ServiceDetailsPage.tsx]: This is reverting of change https://github.com/kiali/kiali-ui/pull/1188/files
Will it keep the logic "when Traces is not configured it should hide Traces tab"?

Comment (aljesusg) [reply src/pages/ServiceDetails/ServiceDetailsPage.tsx]: Yes because we are checking down  if there is a url this.props.jaegerUrl

**Comment by hhovsepy** [comment]
On Maistra, when Traces is correctly configured, it always shows that error message, but in fact traces are available. Just the certs are different:
Screenshot from 2019-05-21 14-34-47
Screenshot from 2019-05-21 14-35-06

**Comment by hhovsepy** [comment]
But the Kiali pod restart fixes that.
> On Maistra, when Traces is correctly configured, it always shows that error message, but in fact traces are available. Just the certs are different:

**Comment by hhovsepy** [comment]
And the opposite, when configuring Traces, than crashing Jaeger pod, only after Kiali pod restart it shows error message that "Could not fetch Traces ...'

**Comment by aljesusg** [comment]
@hhovsepy  Can you test this with the PR https://github.com/kiali/kiali/pull/1089 ?

**Comment by aljesusg** [comment]
Was merged @hhovsepy so please test it with the new backend

**Comment by hhovsepy** [comment]
Tested:
1. Traces pod is killed, error notification is shown and Traces tab is there. So verified previous issues:
Screenshot from 2019-05-23 11-49-02

2. Traces pod killed, traces routing is deleted, Kiali pod is restarted. In this case Traces tab is gone, and another error notification is shown, it did not happen before, but is it expected behavior?:
Screenshot from 2019-05-23 11-41-33

**Comment by aljesusg** [comment]
you're right @hhovsepy we shoudl show the message only if there is a jaeger configured and integration. let me a second

**Comment by aljesusg** [comment]
updated, thanks @hhovsepy

**Comment by hhovsepy** [comment]
Verified:
1. Traces pod kill/start. Error notification is shown/gone.
2. Traces is not configured. No Traces tab and error message.

**Comment by aljesusg** [comment]
ok so @lucasponce @hhovsepy we can review and merge :) Thanks !!!!

**Pull request merged**